src/lib/core/compatibility/compatibility.ts

```diff
@@ -170,14 +172,41 @@ export class CompatibilityModule {
     };
   }
 
-  constructor(@Optional() @Inject(DOCUMENT) document: any) {
-    if (isDevMode() && typeof document && !document.doctype) {
+  constructor(@Optional() @Inject(DOCUMENT) private _document: any) {
+    this._checkDoctype();
+    this._checkTheme();
+  }
+
+  private _checkDoctype(): void {
+    if (isDevMode() && this._document && !this._document.doctype) {
       console.warn(
         'Current document does not have a doctype. This may cause ' +
         'some Angular Material components not to behave as expected.'
       );
     }
   }
+
+  private _checkTheme(): void {
+    if (hasCheckedThemePresence || !this._document || !isDevMode()) {
```

Comment:
Should we move the `isDevMode` check to the constructor now that we're doing multiple checks in the compatibility module?

We can also just have one `hasDoneGlobalChecks` property for all checks instead of `hasCheckedThemePresence`.